5/README.md

```diff
@@ -45,23 +53,72 @@ There are many use cases for the usage of a Property:
 
 * To change the representation of an information in order to be more user friendly a Property can define a new data format for the graphical user interface to convert the raw data.
 
+* TODO: Memorized Attribute
 
-## Specification
-
-A Tango Property is a strict definition of a pair of key/value
-* configuration are represented in the form of properties.
-* data are represented in the form of attributes.
-* actions are represented in the form of commands.
-
-Its model can be represented as a defined tree which each elements are from a defined types: Class, Device, Property, Attribute, Command following the rules below:
 
+## Specification
 
 A Tango Property is a strict definition of a pair of key/value
 * The Property SHALL have one key, called Property Name
-* The Property SHALL have one value
+* The Property SHALL have one value, which could be empty, called Property Value
+* If the device is started upon a Tango Database, Class, Device, Attribute and Free Properties MAY be persisted into the Tango Database (RFC6)
+* If the device is started without a Tango Database, the device and class properties MAY be persisted in a file (but no support for the attribute properties) 
+* All device and class properties MAY be directly modified and accessed through the database device or its file
+* Attribute properties SHOULD be modified and accessed at any time through the device object (RFC-2)
+* All default attribute and device properties are OPTIONAL
+* The attribute, class and device properties MAY be defined in the device code and they MAY be OPTIONAL OR REQUIRED (mandatory)
+* TODO memorized attributes, system properties
+* TODO : properties history
+* TODO: error management
+* All device and class properties MUST be loaded at init command or device start-up
+* If a device property does not exist, the device MUST load the class property having the same name
```

Comment:
The current flow is:
1 - check if class property exist, load it
2 - check if default value exist, load it
3 - check if value is defined in database, load it
This guarantees that, whenever defined, default value will override class value, and db value will override default value (and class as well).
Propose to reword line 74 something like:
"For each Property, the device checks and extracts Property value from Class, default device value and database defined value, with increasing priority"

Reply:
> The current flow is:
> 1 - check if class property exist, load it
> 2 - check if default value exist, load it
> 3 - check if value is defined in database, load it

A more correct way to express it would be:
The current flow is:
1 - check if class property exist, load it, **else** check if default value exist, load it
2 - (executed in any case) check if value is defined in database, load it

The way you wrote it suggested that the default property would override the class property (at least this is the way I interpreted it)

Reply:
> The way you wrote it suggested that the default property would override the class property (at least this is the way I interpreted it)

And it looks like this is the way you interpreted it too.. This is not the case in the current implementations.  A default value does not override a class property.

Reply:
So, starting from your sentence proposal, I would change it to:
_For each Property, the device checks and extracts default device value, Property value from Class and database defined value, with increasing priority._

Reply:
You're right. Had a quick look and missed the "else". But what about "For each Property, the device checks and extracts Property value from default device value, Class and database defined value, with increasing priority" ?

Reply:
> But what about "For each Property, the device checks and extracts Property value from default device value, Class and database defined value, with increasing priority" ?

Looks better to me indeed.

Reply:
What about replacing "database defined value" with "device specific value". Or we should say that  the Device specific value can be defined in multiple places e.g. database, file or in the code (aie!).

Reply:
The way I was reading this is:
default device Property value = if specified in the Pogo dialog... this ends up in the XXXClass.cpp file
default Class Property value = if specified in the Pogo dialog... this ends up in the XXXClass.cpp file
database Class Property value = if Class Property defined in the Tango db... 
database device Property value = if device Property defined in the Tango db...
Not sure about your reference to "file", does this refers to running a device with db on file? Isn't this just a special case of database?

Reply:
To simplify the reading, I would propose to add pseudo-code. Something like:

- A device or class property MAY have an OPTIONAL default value
- The device property loading flow in pseudo-code is:
````
property-value = getDevicePropertyFromTangoDBorFile(device-name, property-name)
if(property-value is empty)
  property-value = getClassPropertyFromTangoDBorFile(class-name, property-name)
if(property-value is empty && default-value is not empty)
  property-value = default-value
```
- The class property loading flow property-name is:
```
property-value = getClassPropertyFromTangoDBorFile(class-name, property-name)
if(property-value is empty && default-value is not empty)
  property-value = default-value`
```